lib/board.rb

```diff
@@ -21,8 +21,68 @@ def initialize
                "D4" => Cell.new("D4")
               }
   end
+  # looping through - nested loop for expansion
 
   def valid_coordinate?(coordinate)
     cells.keys.include?(coordinate)
   end
+
+  def all_same?(array)
+    array.uniq.length == 1 
+  end
+
+  def not_all_same?(array)
+    array.uniq.length > 1 
+  end
+
+  def valid_placement?(ship, cells)
```

Comment:
yes! I think we should convert the letters to numbers/ords here and maybe make this its own helper method

Reply:
great call!